File: src/main/java/ru/johnspade/service/CacheService.java
package ru.johnspade.service;

import org.springframework.stereotype.Service;

import java.util.Date;

@Service
public class CacheService {

	private volatile Date lastSaveTimestamp = new Date();

	public synchronized void setlastSaveTimestamp(Date date) {
		lastSaveTimestamp = date;
	}

	public Date getLastSaveTimestamp() {
		return lastSaveTimestamp;
	}

}

File: src/main/java/ru/johnspade/service/CacheService.kt
package ru.johnspade.service

import org.springframework.stereotype.Service
import java.util.*

@Service
open class CacheService {

	@Volatile internal var lastSaveTimestamp = Date()
		private set

	open @Synchronized internal fun setlastSaveTimestamp(date: Date) {
		lastSaveTimestamp = date
	}

}
